.github/workflows/release.yml

```diff
@@ -5,39 +5,38 @@ on:
     tags: ["v[0-9]+.[0-9]+.[0-9]+*"]
 
 jobs:
-  check:
-    name: Check
+  build:
+    name: Build
     runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        arch: [ "x86_64", "aarch64" ]
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
-      - name: Setup Rust toolchain and cache
-        uses: actions-rust-lang/setup-rust-toolchain@v1.3.7
-      - name: Install just
-        uses: taiki-e/install-action@just
+      - name: Expose git commit data
+        uses: rlespinasse/git-commit-data-action@v1
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        env:
+          RUST_CACHE_KEY_OS: rust-cache-${{ matrix.arch }}-${{ env.GIT_COMMIT_SHORT_SHA }}
```

Comment:
Adding the commit sha to the cache defeats the purpose of the cache at all.
You can either:
* Disable cache with `cache: false`
* Remove the commit sha, and have an arch dependent cache

If you go for option two, if/when we add `musl` support, that should also affect the cache, so it would be good to leave a comment about it.